File: api-dot-net/Dto/ResponseDto/AdminGetProblemDto.cs
using System.Text.Json.Serialization;
using CjsApi.Models;

namespace CjsApi.Dto
{
    public class TestCaseDto
    {
        public string Input { get; set; } = string.Empty;
        public string Output { get; set; } = string.Empty;

        public bool IsSample {get;set;} = false;
    }

    public class AdminGetProblemDto
    {
        public string Title { get; set; } = string.Empty;

        public int Id { get; set; } = 1;
        public string Description { get; set; } = string.Empty;

        public List<TestCaseDto> TestCases { get; set; }
            = new List<TestCaseDto>();

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Difficulty Difficulty { get; set; } = Difficulty.EASY;

        public List<string> Tags { get; set; } = new();
    }
}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/dto/response/AdminGetProblemDto.java
package com.cjs.cjs_service.dto.response;

import com.cjs.cjs_service.dto.TestCaseDto;
import com.cjs.cjs_service.model.Difficulty;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class AdminGetProblemDto {

    private String title = "";

    private int id = 1;

    private String description = "";

    private List<TestCaseDto> testCases = new ArrayList<>();

    private Difficulty difficulty = Difficulty.EASY;

    private List<String> tags = new ArrayList<>();

    // getters and setters
}
